TeamCityQueueScraper.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Prometheus.Client;
+using TeamCitySharp;
+
+namespace TeamCityBuildStatsScraper
+{
+    internal class TeamCityQueueScraper: IHostedService, IDisposable
+    {
+        private readonly IHost _host;
+        private readonly IConfiguration _configuration;
+        private Timer _timer;
+
+        public TeamCityQueueScraper(IHost host, IConfiguration configuration)
+        {
+            _host = host;
+            _configuration = configuration;
+        }
+        
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            // Fire off the Scraper starting *right now* and do it again every minute
+            _timer = new Timer(ScrapeQueueStats, null, TimeSpan.Zero, TimeSpan.FromMinutes(1));
+
+            return Task.CompletedTask;
+        }
+
+        private void ScrapeQueueStats(object state)
+        {
+            var metricFactory = _host.Services.GetRequiredService<IMetricFactory>();
```

Comment:
You should pass in the `IMetricFactory` in the ctor, rather than the `IHost`.
Doing it this way is called "service locator" which is considered icky